src/SpeedyTransforms/spectral_transform.jl

```diff
@@ -331,7 +330,7 @@ function gridded!(  map::AbstractGrid{NF},                      # gridded output
 
         # Recalculate or use precomputed Legendre polynomials Λ
         recompute_legendre && Legendre.unsafe_legendre!(Λw, Λ, lmax, mmax, Float64(cos_colat[j_north]))
-        Λj = recompute_legendre ? Λ : Λs[j_north]
+        Λj = recompute_legendre ? LowerTriangularMatrix(Λ) : Λs[j_north]
```

Comment:
I believe this allocates for a T31 transform 24 T31 matrices -- not ideal!

Reply:
Yeah, absolutely, not ideal in case `recompute_legendre == true`. 

I didn't go very deep into AssociatedLegendrePolynomials.jl, but it does repeatably use `ndims` of the array it's given, even the actual compute routines. So, we can use the `LowerTriangularMatrix` there completely easily. 

I didn't test what happens if you redefine `ndims`, but I expect this to be cause some troubles somewhere else. 

The first quite "hacky" solution that comes to my mind to solve it would be to define another type that is just used for the transforms: 

```Julia
struct LegendePolArray{T,N,ArrayType} <: AbstractArray{T,N}
   data::LowerTriangularArray{T,N-1,ArrayType}
end 

getindex(L:: LegendePolArray, I...) = getindex(L.data, I...)
setindex!(L:: LegendePolArray, x, I...) = setindex!(L.data, x, I...)
```
Broadcast and some other operations wouldn't work for it, but as far as I can see it, everything in `SpectralTransforms` would. 

But yeah, that is a bit hacky as a solution to that problem.

Reply:
I implemented it like this now